compiler/desugared/name_resolution.mli

```diff
@@ -162,6 +162,10 @@ val get_def_key :
   Ast.ScopeDef.t
 (** Usage: [get_def_key var_name var_state scope_uid ctxt pos]*)
 
+val surface_rule_to_def : Surface.Ast.rule -> Surface.Ast.definition
+(** Translates a {!type: Surface.rule} into the corresponding
+    {!type: Surface.definition} *)
+
```

Comment:
Why moving this here? Dependency cycle?

Reply:
Since removing the use of `Visitors`, it became possible to make `Surface.Ast` an interface-only module by just removing these two functions.